src/complexity_measures/statistical_complexity.jl

```diff
@@ -0,0 +1,192 @@
+using Distances
+
+export StatisticalComplexity, maximum_complexity_entropy, minimum_complexity_entropy
+
+evaluate(distance::SemiMetric, a::Probabilities, b::Probabilities) = Distances.evaluate(distance, a.p, b.p)
+
+"""
+    StatisticalComplexity <: ComplexityMeasure
+    StatisticalComplexity(est = SymbolicPermutation(),
+                          distance = JSDivergence(),
+                          entropy = Renyi())
+
+Calculate the statistical complexity of a time series `x` as defined by Rosso et al. [^Rosso2007], based on 
+permutation patterns.
+
+## Description
+The statistical complexity ``C_{JS}[P]`` is defined based on the distribution ``P`` of 
+ordinal patterns
+```math
+C_{JS}[P] = Q_J[P, P_e] H_S[P]
+```
+where ``Q_J`` is originally the normalized Jensen-Shannon divergence of the ordinal distribution ``P`` to 
+a uniform distribution ``P_e`` with the same numer of bins, and ``H_S`` is the normalized permutation entropy, 
+`entropy_normalized(x, SymbolicPermutation())`.
+In this implementation, the distance measure defaults to the Jensen-Shannon divergence, but other (semi-)metrics can also be
+chosen. The type of entropy is also a free parameter, but defaults to the (originally used) Renyi entropy.
+
+For an ordinal distribution with a given permutation entropy, there is a minimum and maximum value that the
+statistical complexity can take. The minimum and maximum statistical complexity as a function of the entropy
+is calculated using [`minimum_complexity_entropy`](@ref) and [`maximum_complexity_entropy`](@ref).
+
+Standard usage:
+```julia
+m, τ = 3, 1
+est = SymbolicPermutation(; m, τ)
+c = StatisticalComplexity(; est, entropy=Renyi())
+x = randn(100)
+statistical_complexity = complexity(c, x)
+```
+
+NOTE: I'm not sure if this API is congruent with the general complexity API, but I think
+that, since the "standard" signature is also possible, it should be fine.
+I have implemented the other signature because I find it redundant to calculate the entropy
+twice, which is needed for the complexity anyway and usually calculated in combination with it.
+
+Because the distributions of permutation symbols are needed both for the permutation entropy and the
+statistical complexity, the same can also be achieved using
+```julia
+p = probabilities(est, x)
+permutation_entropy = entropy(est, x)
+statistical_complexity = complexity(c, p; h=permutation_entropy)
+```
+
+[^Rosso2007] Rosso, O. A., Larrondo, H. A., Martin, M. T., Plastino, A., & Fuentes, M. A. (2007). 
+            [Distinguishing noise from chaos](https://doi.org/10.1103/PhysRevLett.99.154102).
+            Physical review letters, 99(15), 154102. 
+"""
+Base.@kwdef struct StatisticalComplexity <: ComplexityMeasure
+    est::ProbabilitiesEstimator = SymbolicPermutation()
+    distance::SemiMetric = JSDivergence()
+    entropy::Entropy = Renyi()
+end
```

Comment:
All of this is type unstable. Code using `StatisticalComplexity` is type unstable and will have bad performance. What you want to do instead is:

```suggestion
Base.@kwdef struct StatisticalComplexity{P<:ProbabilitiesEstimator, M, E <: Entropy} <: ComplexityMeasure
    est::P= SymbolicPermutation()
    distance::M = JSDivergence()
    entropy::E = Renyi()
end
```